### DISCUSSION TITLE
Trying to disable CHECK_LOAD using extraEnvVars in values file passed to helm chart

### DISCUSSION THREAD

**Discussion by alarys** [MoJo2600/pihole-kubernetes#222] [General]
Hi,

I am getting 15 minute load average warnings from my pihole deployed to a k3s cluster using this helm chart.

I know it is benign, as pihole is working well, and my load monitoring shows that my cluster is doing fine for load. Other people concur and say you can disable the load check in the /etc/pihole/pihole-FTL.conf file with:

https://docs.pi-hole.net/ftldns/configfile/#check_load

However, since I am using the helm chart here, I tried to add this to the values yaml file I am using. I noted that in the chart options https://github.com/MoJo2600/pihole-kubernetes/blob/master/charts/pihole/values.yaml there is a variable for extraEnvVars:

```
# -- extraEnvironmentVars is a list of extra enviroment variables to set for pihole to use
extraEnvVars: {}
  # TZ: UTC
```

So I added this section to my helm chart values yaml file

```
extraEnvVars: 
  CHECK_LOAD: false
```

But when I upgrade my helm install, the /etc/pihole/pihole-FTL.conf doesn't contain the extra variable and the log file /var/log/pihole-FTL.log says that CHECK_LOAD is still enabled.

`[2022-03-13 16:49:28.243 483M]    CHECK_LOAD: Enabled`

Any help would be appreciated. I am sure there are other people using this excellent Helm chart to deploy pihole, and would really like to reduce the false alarms from the LOAD_CHECK.

**Comment by MoJo2600** [comment]
I'm sorry, I tested it on my cluster and it seems to work as expected.

I updated my `values.yaml`
```
extraEnvVars:
  CHECK_LOAD: false
```

Checked the helm template I generated with ` helm upgrade -f values.yaml -i pihole charts/pihole/`
```
      containers:
        - name: pihole
          env:
          ...
          - name: "CHECK_LOAD"
            value: "false"
```

and the pod looks like this:
```
    Environment:
      ...
      CHECK_LOAD:    false
      ...
```
inside the container: 

```
root@pihole-746cc66d7b-h46cw:/# env | grep -i CHECK
CHECK_LOAD=false
```

Can we maybe double check if you are using the latest version?

Im using:
```
helm list
NAME    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART           APP VERSION
pihole  dns             10              2022-03-14 11:06:13.116510028 +0100 CET deployed        pihole-2.5.8    2022.02.1
```

**Reply by alarys** [reply]
Hi there,

Thanks, this was very helpful. 

I did verify that yes, the env variable is indeed being set for me (I thought it would be added to one of the configuration files, which is where I was looking, knowing now to check the env setting is helpful).

`# env | grep -i CHECK
CHECK_LOAD=false`

I checked my pihole version and it was behind.

`helm list -n pihole
NAME    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART           APP VERSION
pihole  pihole          1               2022-03-14 10:38:35.638568042 -0700 PDT deployed        pihole-2.5.7    2022.02.1`

I updated it so now I'm on the same version as you.

`helm list -n pihole
NAME    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART           APP VERSION
pihole  pihole          2               2022-03-14 11:01:27.713057095 -0700 PDT deployed        pihole-2.5.8    2022.02.1`

I uninstalled my pihole helm install, deleted the storage, and did the deploy again. For sure the env is being set correctly.

When I look at the logs, it says that CHECK_LOAD is enabled?

`# grep LOAD /var/log/pihole-FTL.log
[2022-03-14 18:01:40.560 489M]    CHECK_LOAD: Enabled`

**My question is does the env variable over-ride this? It's a bit confusing to me that the log says that is it enabled, but the env says that it is disabled. Does that make sense?**

Thanks!

**Reply by alarys** [reply]
Hi,

I got my answer unfortunately. While the CHECK_LOAD=false is being set in the env variable, I've confirmed that it is still enabled in the logs, and on my pi-hole web interface, I got a message about LOAD.

`2022-03-16 12:11:40 | LOAD | Long-term load (15min avg) larger than number of processors: 4.1 > 4This may slow down DNS resolution and can cause bottlenecks.`

Is there anything else I can do to disable the LOAD messages?

Many thanks in advance :-) Pi-hole is great, and I can't imagine how much work it is. This is a pretty minor issue.
